scipy/stats/stats.py

```diff
@@ -7469,6 +7470,214 @@ def brunnermunzel(x, y, alternative="two-sided", distribution="t",
+
+    .. math::
+
+        \mathbb{P}(x_m \leq q) = \mathbb{P}(x_{N-m+1} \geq 1-q).
+
+    The function computes confidence intervals for a given quantile and
+    confidence level, based on `x` which is either a set of samples
+    (one-dimensional array_like) or the number of samples available.
+    The confidence intervals are valid if and only if the samples are i.i.d.
+
+    Both one-sided and two-sided confidence intervals can be obtained
+    (default is one-sided). The function returns two values: either the bounds for the two one-sided
+    confidence intervals, or the lower and upper bounds of a two-sided confidence interval.
+    The return values are either the indexes of the bounds (if `x` is an integer) or
+    sample values (if `x` is the set of samples).
+    `None` is returned when there are not enough samples to compute
+    the desired confidence interval.
+
+    There is no uniqueness of the two-sided confidence interval (see Notes below).
+    Without further assumption on the samples (eg, the nature of the underlying distribution),
+    the one-sided intervals are optimally tight.
+
+    Parameters
+    ----------
+    x : array_like or int
+        Array of samples, should be one-dimensional.
+        If integer, taken as the number of samples available (strictly positive)
+    quantile : float
+        The quantile for which we want to compute the confidence interval.
+        Must be strictly between 0 and 1.
+    confidence : float
+        The desired confidence level of the confidence interval.
+        Must be strictly between 0 and 1.
+    type : {'one-sided', 'two-sided'}, optional
+        Defines the type of confidence interval computed.
+        Default is 'one-sided'.
+
+          * 'one-sided' : computes the best possible one-sided confidence intervals (both lower and upper bounds) for the given quantile.
+          * 'two-sided' : computes a two-sided confidence interval by combination of two one-sided intervals. E.g., a 90% two-sided interval is computed by combining two 95% one-sided intervals
+    verbose : bool, optional
+        Enable verbose outputs, default is False
+
+    Returns
+    -------
+    LB : float or int
+        value or index of the lower bound of
+
+        * the right-open one-sided confidence interval (default, ``type=one-sided``),
+        * a two-sided confidence interval (if ``type=two-sided``)
+    UB : float or int
+        value or index of the upper bound of
+
+        * the left-open one-sided confidence interval (default, ``type=one-sided``),
+        * a two-sided confidence interval (if ``type=two-sided``)
+
+    Notes
+    -----
+    Two-sided confidence intervals are not guaranteed to be optimal.
+    I.e., there may exist a tighter interval that may contain the quantile
+    of interest with probability larger than the confidence level.
+    These intervals may be found by exhaustive search,
+    which we do not do for efficiency reason.
```

Comment:
"for reasons of efficiency" maybe

Reply:
I'm not an English native speaker, "for efficiency reason" sounds more idiomatic to me, but I may be wrong. Although it's written more in plural apparently (for efficiency reason**s**). Will update that.